src/core/baseClient.ts

```diff
@@ -302,4 +305,49 @@ export class BaseClient {
       )
     );
   }
+  /**
+   * Gets the current student's rewards shop
+   * @returns Array of purchasable items
+   */
+  async getRewards(): Promise<RewardsResponse> {
+    return (
+      await this.makeAuthedRequest(
+        this.API_BASE + "/rewards/" + this.studentId,
+        {
+          method: "GET",
+        },
+      )
+    );
+  }
+  /**
+   * Purchase a reward item from the current student's  rewards shop
+   * @param itemId number
+   * @returns An object containg balence and id
+   */
+  async postRewardPurchase(itemId: number): Promise<RewardPurchaseResponse> {
```

Comment:
```suggestion
  async purchaseReward(itemId: number): Promise<RewardPurchaseResponse> {
```
Perhaps this method would be better named `purchaseReward`. Sorry if it was unclear but in the other methods get wasn't meaning the http method, it was simply a verb.